src/arithmetic/point_funcs/point_funcs.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2018-2020 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#include "RG.h"
+#include "../func_desc.h"
+#include "../../errors.h"
+#include "../../util/arr.h"
+#include "../../datatypes/map.h"
+#include <math.h>
+
+#define EARTH_RADIUS 6378140.0
+#define DegreeToRadians(d) ((d) * M_PI / 180.0)
+
+SIValue AR_TOPOINT(SIValue *argv, int argc) {
+	SIValue map = argv[0];
+	SIType t = SI_TYPE(map);
+
+	if(t == T_NULL) return SI_NullVal();
+
+	// expecting input to be a map
+	// point({latitude: 32.0705767, longitude: 34.8185946})
+	ASSERT(t == T_MAP);
+
+	uint key_count = Map_KeyCount(map);
+	if(key_count != 2) {
+		// TODO: error
+	}
+
+	bool key_exists;
+	SIValue latitude;
+	SIValue longitude;
+
+	if(!Map_Get(map, SI_ConstStringVal("latitude"), &latitude)) {
+		// TODO: error
+	}
+	if(!Map_Get(map, SI_ConstStringVal("longitude"), &longitude)) {
+		// TODO: error
+	}
+
+	if(SI_TYPE(latitude) != T_DOUBLE || SI_TYPE(longitude) != T_DOUBLE) {
+		// TODO: error
+	}
+
+	float lat = latitude.doubleval;
+	float lon = longitude.doubleval;
+	return SI_Point(lat, lon);
+}
+
+SIValue AR_DISTANCE(SIValue *argv, int argc) {
+	// compute distance between two points
+	// a = sin²(Δφ/2) + cos φ1 ⋅ cos φ2 ⋅ sin²(Δλ/2)
+	// c = 2 * atan2( √a, √(1−a) )
+	// d = R * c
+	// where φ represent the latitudes, and λ represent the longitudes
+
+	SIValue p1 = argv[0];
+	SIValue p2 = argv[1];
+
+	// check inputs
+	if(SI_TYPE(p1) == T_NULL || SI_TYPE(p2) == T_NULL) return SI_NullVal();
+
+	float lat[2] = { DegreeToRadians(p1.point.latitude),
+		DegreeToRadians(p2.point.latitude) };
+
+	float lon[2] = { DegreeToRadians(p1.point.longitude),
+		DegreeToRadians(p2.point.longitude) };
+
+	float dlat = lat[1] - lat[0];
+	float dlon = lon[1] - lon[0];
+
+	// a = sin²(Δφ/2) + cos φ1 ⋅ cos φ2 ⋅ sin²(Δλ/2)
+	float a = pow(sin(dlat / 2), 2) + cos(lat[0]) * cos(lat[1]) * pow(sin(dlon / 2), 2);
+
+	// c = 2 * atan2( √a, √(1−a) )
+	float c = 2 * atan2(sqrt(a), sqrt(1-a));
```

Comment:
This step is different than the one Wikipedia lists for the Haversine formula. Is it a different formula, or are they equivalent expressions?